main.go

```diff
@@ -213,15 +221,111 @@ func fetchOrCreateTestPlan(ctx context.Context, cfg config.Config, files []strin
 		}
 		// Create the fallback plan
 		fmt.Println("Could not fetch plan from server, using fallback mode. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(testCases, cfg.Parallelism)
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
 	}
 
 	// The server can return an "error" plan indicated by an empty task list (i.e. `{"tasks": {}}`).
 	// In this case, we should create a fallback plan.
 	if len(testPlan.Tasks) == 0 {
 		fmt.Println("Test splitter server returned an error, using fallback mode. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(testCases, cfg.Parallelism)
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
 	}
 
 	return testPlan, nil
 }
+
+type fileTiming struct {
+	Path     string
+	Duration time.Duration
+}
+
+// createRequestParam creates the request parameters for the test plan.
+// If SplitByExample is disabled (default), it will return the default params that contain all the files.
+// If SplitByExample is enabled, it will split the slow files into examples and return it along with the rest of the files.
+//
+// Error is returned if there is a failure to fetch test file timings or to get the test examples from test files when SplitByExample is enabled.
+func createRequestParam(cfg config.Config, files []string, client api.Client, runner TestRunner) (api.TestPlanParams, error) {
+	if !cfg.SplitByExample {
+		debug.Println("Splitting by file")
+		testCases := []plan.TestCase{}
+		for _, file := range files {
+			testCases = append(testCases, plan.TestCase{
+				Path: file,
+			})
+		}
+
+		return api.TestPlanParams{
+			Mode:        cfg.Mode,
+			Identifier:  cfg.Identifier,
+			Parallelism: cfg.Parallelism,
+			Tests: api.TestPlanParamsTest{
+				Files: testCases,
+			},
+		}, nil
+	}
+
+	debug.Println("Splitting by example")
+
+	debug.Printf("Fetching timings for %d files", len(files))
+	// Fetch the timings for all files.
+	timings, err := client.FetchFilesTiming(cfg.SuiteSlug, files)
+	if err != nil {
+		return api.TestPlanParams{}, fmt.Errorf("failed to fetch file timings: %v", err)
+	}
+	debug.Printf("Got timings for %d files", len(timings))
+
+	// The server only returns timings for the files that has been run before.
+	// Therefore, we need to merge the response with the requested files.
+	// The files that are not in the response will have a duration of 0.
+	allFilesTiming := []fileTiming{}
+	for _, file := range files {
+		allFilesTiming = append(allFilesTiming, fileTiming{
+			Path:     file,
+			Duration: timings[file],
+		})
+	}
+
+	// Sort the files by duration.
+	// If the duration is the same, sort by path.
+	slices.SortFunc(allFilesTiming, func(a, b fileTiming) int {
+		if a.Duration == b.Duration {
+			return strings.Compare(a.Path, b.Path)
+		}
+		return int(b.Duration - a.Duration)
+	})
+
+	// Split the timings into slow files and the rest of the files.
+	threshold := int(float64(len(allFilesTiming)) * 0.3)
```

Comment:
Hmmm, I wonder if we should be a bit smarter than just taking the slowest 30%? 

Like consider these scenarios:
- the changeset add lots of new files which we won't have timings for. there are so many new files that some of them are sorted in the slowest 30%
- the test files are all really small. perhaps we should apply a minimum, say 60s for a file's execution time before it meets the slow threshold? 

Like instead of having to SBE the slowest 30%, maybe it would be better to split *up to* 30% of the slowest test files?

Reply:
This is a good point. Maybe we can split the 30% of _known_ slow files instead? The `GET /test_files` endpoint only returns test files that have been executed before, so we can be confident that the new test files isn't included in the 30% slow files? What do you think?

Reply:
As per slack discussion - I reckon we change so that we only split files that take longer than 3min to execute!

Reply:
We discussed this in Slack, and decided to define "slow files" as files that have duration greater or equal to 3 minutes.